app/src/main/java/com/nutomic/syncthingandroid/activities/FolderActivity.java

```diff
@@ -480,6 +480,7 @@ private void addEmptyDeviceListView() {
     private void addDeviceViewAndSetListener(Device device, LayoutInflater inflater) {
         inflater.inflate(R.layout.item_device_form, mDevicesContainer);
         SwitchCompat deviceView = (SwitchCompat) mDevicesContainer.getChildAt(mDevicesContainer.getChildCount()-1);
+        deviceView.setOnCheckedChangeListener(null);
```

Comment:
Why would you not do it why all other deregistrations are done?

Reply:
If you compare it to the other AddListeners function, it's the same way of successfully suppressing the event when we (initially) change the slider from code. There is no onDestroy() deregistration of the listeners in place as it's not necessary to do so. (SettingsActivity also uses the the above approach...) Otherwise, we would need a boolean in the class' context to be set true if we fire the event by code instead of the UI triggering it.

Reply:
So this essentially removes the listener that was previously registered? Why don't re unset it as the view gets destroyed?

Unsetting on creation feels wrong.

Reply:
I think it's the function name I kept from the existing code that is quite confusing. This function is called whenever the UI is refreshed, so it's not a unset on creation. To be more precise, we don't kill a listener here but the function that fires if the event occurs. So that's okay, just believe me :)